web/packages/teleterm/src/ui/DocumentCluster/ActionButtons.tsx

```diff
@@ -180,3 +190,77 @@ async function getDatabaseUsers(appContext: IAppContext, dbUri: DatabaseUri) {
     throw e;
   }
 }
+
+function AppButton(props: {
+  isWebApp: boolean;
+  launchUrl: string;
+  connect(): void;
+  onLaunchUrl(): void;
+}) {
+  const ref = useRef<HTMLButtonElement>();
+  const [isOpen, setIsOpen] = useState(false);
+
+  if (!props.isWebApp) {
+    return (
+      <ButtonBorder size="small" onClick={props.connect}>
+        Set up connection
+      </ButtonBorder>
+    );
+  }
+
+  return (
+    <Flex>
+      <ButtonBorder
+        size="small"
+        forwardedAs="a"
+        href={props.launchUrl}
+        onClick={props.onLaunchUrl}
+        target="_blank"
+        title="Launch app in a browser"
+        css={`
+          border-top-right-radius: 0;
+          border-bottom-right-radius: 0;
+        `}
+      >
+        Launch
+      </ButtonBorder>
+      <ButtonBorder
```

Comment:
If I open the menu and then close it, the background color of this button remains the same as when the menu was open. I need to make an additional click for the color to reset.

Though I see that it's probably a universal problem with how our Button and Menu components work together. The Connect button for other resources behaves in the same way.

Reply:
I think `Menu` has a mechanism that returns focus to the last focused element which changes the background color.